### ISSUE TITLE
R0917 incorrectly counts number of positional arguments

### ISSUE THREAD

**Issue by ddeepwell** [pylint-dev/pylint#9996]
### Bug description

Pylint raises a too-many-positional-arguments when there are too many arguments, but not too many positional arguments. This unexpected behaviour was introduced in pylint version 3.3.0. 

Given the file, `file.py`:

```python
"""Function."""

def function(a, b, c, d, e="", f=""):
    """Print stuff."""
    print(a, b, c, d, e, f)
```

There are 4 positional arguments, but pylint complains that there are 6.

### Command used

```shell
pylint file.py
```

### Pylint output

```python
************* Module file
file.py:3:0: R0913: Too many arguments (6/5) (too-many-arguments)
file.py:3:0: R0917: Too many positional arguments (6/5) (too-many-positional-arguments)
```

### Expected behavior

I expect to see:

```
************* Module file
file.py:3:0: R0913: Too many arguments (6/5) (too-many-arguments)
```

### Pylint version

```shell
pylint 3.3.1
astroid 3.3.4
Python 3.12.4 (main, Jun  6 2024, 18:26:44) [Clang 15.0.0 (clang-1500.3.9.4)]
```

### OS / Environment

MacOS 14.6.1

**Comment by jacobtylerwalls** [comment]
See the documentation for this message for more details and examples. "positional" is really a shorthand for "positional-or-keyword", which is a mouthful. Your message goes away if you insert "*" to make the `e` and `f` args keyword only, which is the point of this message.

**Comment by Pierre-Sassoulas** [comment]
(https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/too-many-positional-arguments.html)
